### DISCUSSION TITLE
Git add . staging ganache node modules

### DISCUSSION THREAD

**Discussion by zbilly18** [smartcontractkit/full-blockchain-solidity-course-py#1668] [Q&A] [answered]
After executing git add . and git status, I am seeing that there are 1k+ staged changes in the source control. Not sure what is happening here but I can confirm that my source code is in the git add along with many other files.

![image](https://user-images.githubusercontent.com/65373914/186755254-ac23830d-e714-4068-a433-be94c6c2fd13.png)

**Comment by cromewar** [comment, marked as answer]
Hello @zbilly18 
Well this is because you are also adding the `node_modules` folder, which contains hundreds if not thousands of dependencies for the project to run (if you have installed something with npm).
So, is as easiest as go to `.gitignore` file and add `node_modules/` to it, and now git will ignore it.

Note: if you don't have a node_modules on your project file, you might have it on a parent folder upside the hierarchy.

Cheers.
